src/pybind/mgr/rbd_support/mirror_snapshot_schedule.py

```diff
@@ -700,6 +700,10 @@ def remove_schedule(self,
             "remove_schedule: level_spec={}, interval={}, start_time={}".format(
                 level_spec.name, interval, start_time))
 
+        if interval:
+            match = re.match(r'^(\d+)(d|h|m)?$', interval)
+            if not match:
+                raise ValueError("Invalid value for interval ({})".format(interval))
```

Comment:
I would just use `Interval.from_string(interval)` for validation instead of manual regex matching.

And I believe the same problem exists in trash_purge_schedule.py.
So I wonder, wouldn't it better to have this validation in `Schedules.remove`. I would just have something like this there:
```
    def remove(self,
               level_spec: LevelSpec,
               interval_str: Optional[str],
               start_time_str: Optional[str]) -> None:
  interval = Interval.from_string(interval_str)
  start_time =  StartTime.from_string(start_time_str)
  ...
    schedule.remove(interval, start_time)
```
Or something else if you wish. E.g. your previous version with `self.schedules.pop` would work too, if you improved it  to  do `del self.schedules[level_spec.id]` if the `schedule` becomes empty after the interval removal.

Reply:
I missed trash_purge, thanks! update patch.